adapters/zio-http/src/main/scala/caliban/ZHttpAdapter.scala

```diff
@@ -3,27 +3,36 @@ package caliban
 import caliban.interop.tapir.ws.Protocol
 import caliban.interop.tapir.{ HttpInterpreter, WebSocketInterpreter }
 import sttp.capabilities.zio.ZioStreams
+import sttp.model.HeaderNames
 import sttp.tapir.server.ziohttp.{ ZioHttpInterpreter, ZioHttpServerOptions }
 import zio.http._
 
 object ZHttpAdapter {
 
+  @deprecated("Defining subprotocols in the server config is no longer required")
   val defaultWebSocketConfig: WebSocketConfig = {
     val subProtocols = List(Protocol.Legacy.name, Protocol.GraphQLWS.name).mkString(",")
-    WebSocketConfig.default.withSubProtocol(Some(subProtocols))
+    WebSocketConfig.default.subProtocol(Some(subProtocols))
   }
 
   def makeHttpService[R, E](interpreter: HttpInterpreter[R, E])(implicit
     serverOptions: ZioHttpServerOptions[R] = ZioHttpServerOptions.default[R]
-  ): App[R] =
+  ): HttpApp[R] =
     ZioHttpInterpreter(serverOptions)
       .toHttp(interpreter.serverEndpoints[R, ZioStreams](ZioStreams))
-      .withDefaultErrorResponse
 
   def makeWebSocketService[R, E](interpreter: WebSocketInterpreter[R, E])(implicit
     serverOptions: ZioHttpServerOptions[R] = ZioHttpServerOptions.default[R]
-  ): App[R] =
-    ZioHttpInterpreter(serverOptions)
-      .toHttp(interpreter.serverEndpoint[R])
-      .withDefaultErrorResponse
+  ): HttpApp[R] = {
+
+    val opts = serverOptions.withCustomWebSocketConfig { req =>
+      val protocol = req.header(HeaderNames.SecWebSocketProtocol).fold(Protocol.Legacy: Protocol)(Protocol.fromName)
+      serverOptions.customWebSocketConfig(req) match {
+        case Some(existing) => existing.subProtocol(Some(protocol.name))
+        case _              => WebSocketConfig.default.subProtocol(Some(protocol.name))
+      }
+    }
```

Comment:
I decided to patch the user-provided one instead of making the one that handles subprotocols the default. This way users that override the default server options won't have to do make changes to their code